### ISSUE TITLE
Testdrive schemas seem to be whitespace sensitive

### ISSUE THREAD

**Issue by umanwizard** [MaterializeInc/materialize#2428]
For example, this command:

```
$ set trxschema={
  "name": "trx",
  "type": "record",
  "fields":
  {"name": "id", "type": "string"},
  {"name": "status", "type": "string"}
  ]
  }
```

produces the error:

```
==> test/testdrive/basic.td
test/testdrive/basic.td:28:31: error: command argument has unterminated open brace
     |
  27 |   {"name": "id", "type": "string"},
  28 |   {"name": "status", "type": "string"}
     |                               ^
```

Whereas this command, exactly the same but with different formatting, works fine:

```
$ set trxschema={
    "name": "trx",
    "type": "record",
    "fields": [
      {"name": "id", "type": "string"},
      {"name": "status", "type": "string"}
    ]
  }
```

**Comment by umanwizard** [comment]
@nacrooks and I have both been burned by this.

**Comment by benesch** [comment]
Your first example is not syntactically valid. It's missing an open square brace after "fields". If you fix that, it works fine.

**Comment by nacrooks** [comment]
Tabs vs whitespaces cause an issue for me.

**Comment by umanwizard** [comment]
@benesch Sorry, you're right. Still, there are other sources of confusion: for example, having the close brace on the first column will cause a parse error.

Also, the error message doesn't point at the right place.

**Comment by wangandi** [comment]
I'll take this since I recently changed testdrive brace/bracket handling. @nacrooks Is there an example of tabs vs. whitespaces causing an issue that I can try out?

**Comment by benesch** [comment]
> for example, having the close brace on the first column will cause a parse error.

Yes, testdrive is whitespace sensitive, and the first column is special. A `$` means new command, a `>` means new SQL command, a `!` means new fail SQL command, and a ` ` means a continuation of the last non ` `.  Tabs are not expected to work. 

Certainly it would be nice if the error messages were better! But the parser, especially the position tracker, are rather a mess, and probably should just be rewritten with a proper lexer. I don't think we should spend much time on this right now. There are a lot of other pressing issues.
